src/components/TimePicker/TimePicker.jsx

```diff
@@ -1,46 +1,185 @@
 // @flow
-import React, { Component } from 'react';
+import moment from 'moment';
+import React, { type Node } from 'react';
+import cn from 'classnames';
+import ReactSelect from 'react-select';
+import 'react-select/dist/react-select.min.css';
 import style from './style.scss';
-import type { Data } from '../../types';
+import Text from '../Text/Text';
+
+/**
+ * A TimePicker component.
+ *
+ * Optional 'value' must be in 24H 'HH:mm' format.
+ *
+ * Optional 'minTime' in 24H 'HH:mm' format, default time 08:00.
+ *
+ * Optional 'maxTime' in 24H 'HH:mm' format, default time 20:00.
+ *
+ * Optional 'timeIntervalMinutes', Time step for time slots, default is 30 minute.
+ *
+ * Optional 'timeFormatType' label type in 24H or 12H, default 24H.
+ *
+ * Optional 'defaultOption':
+ *    nextInterval = will round up: 11:26 -> 11:30  based on 'timeIntervalMinutes'.
+ *    minimum (default selection) =  will select 'minTime'.
+ *
+ * Optional 'disabled'.
+ *
+ * OnChange receives string value in 24H 'HH:mm' format.
+ */
+
+export const FormatTypes = Object.freeze({
+  timeFormat12: 'hh:mm A',
+  timeFormat24: 'HH:mm',
+});
+
+export type TimeFormatType = $Values<typeof FormatTypes>;
+
+export const DefaultOptions = Object.freeze({
+  minimum: 'minimum',
+  nextInterval: 'nextInterval',
+});
+
+export type DefaultOption = $Values<typeof DefaultOptions>;
+
+type Option = {|
+  label: string,
+  value: string,
+|};
 
 type Props = {|
-  onBlur: (event: { target: { value: Object } }) => void,
-  onChange: (event: { target: { value: Object } }) => void,
-  time: string,
-  data: Data,
+  minTime: string,
+  maxTime: string,
+  timeIntervalMinutes: number,
+  timeFormat: TimeFormatType,
   disabled: boolean,
+  value?: string,
+  onChange: string => void,
+  className?: string,
+  placeholder?: string,
+  defaultOption?: DefaultOption,
 |};
 
-class TimePicker extends Component<Props> {
-  static displayName = 'Plasma@TimePicker';
+type State = {|
+  value: string,
+|};
 
-  handleBlur = (event: { target: { value: Object } }): void => {
-    const newTime = event.target.value;
+class TimePicker extends React.Component<Props, State> {
+  static defaultProps = {
+    minTime: '08:00',
```

Comment:
Defaults should probably from 00:00 to 23:59 or something like that. 8 to 20 is pretty random.

Reply:
we work office hours are 8-20

Reply:
It varies, think it would be better if it will be customable as well

Reply:
its optional prop the default is 8-8, you can change it by passing minTime / maxTime

Reply:
agree with @marister on that, it's a general time picker component, not coupled specifically to conference rooms